core/src/androidMain/kotlin/Bluetooth.kt

```diff
@@ -38,10 +41,21 @@ private val Context.locationManager: LocationManager
     get() = ContextCompat.getSystemService(this, LocationManager::class.java)
         ?: error("LocationManager system service unavailable")
 
-private val isLocationEnabled: Boolean
-    get() = LocationManagerCompat.isLocationEnabled(applicationContext.locationManager)
+private fun isLocationEnabled(): Boolean {
+    return LocationManagerCompat.isLocationEnabled(applicationContext.locationManager)
+}
 
-internal actual val bluetoothAvailability: Flow<Bluetooth.Availability> =
+private val locationNotNeededStateFlow: Flow<Boolean> =
+    when {
+        SDK_INT > R -> flow { emit(true) }
+        else -> broadcastReceiverFlow(IntentFilter(LocationManager.PROVIDERS_CHANGED_ACTION))
+            .map { intent ->
+                intent.getBooleanExtra(LocationManager.EXTRA_PROVIDER_ENABLED, false)
```

Comment:
Unfortunately [`LocationManager.EXTRA_PROVIDER_ENABLED`](https://developer.android.com/reference/android/location/LocationManager#EXTRA_PROVIDER_ENABLED) requires API 30.

Being that we don't have to check location services status after API 30, this really only adds any value for Android 11 (API 30). If we're below API 30, then we unfortunately have to fallback to just getting the single emission from `onStart`. 😢 

If you think that having the updates is valuable (even if only available on a single additional Android version), then I believe the following would work?:

```kotlin
private val isLocationEnabled: Boolean
    get() = LocationManagerCompat.isLocationEnabled(applicationContext.locationManager)

private val locationEnabledFlow = when {
    SDK_INT > R -> flowOf(true)
    SDK_INT == R -> broadcastReceiverFlow(IntentFilter(PROVIDERS_CHANGED_ACTION))
        .map { intent -> intent.getBooleanExtra(EXTRA_PROVIDER_ENABLED, false) }
        .onStart { emit(isLocationEnabled) }
        .distinctUntilChanged()
    else -> flowOf(isLocationEnabled)
}

private val bluetoothStateFlow =
    broadcastReceiverFlow(IntentFilter(BLUETOOTH_STATE_CHANGED))
        .map { intent -> intent.getIntExtra(EXTRA_STATE, ERROR) }
        .map { state ->
            when (state) {
                STATE_ON -> Available
                STATE_OFF -> Unavailable(reason = Off)
                STATE_TURNING_OFF -> Unavailable(reason = TurningOff)
                STATE_TURNING_ON -> Unavailable(reason = TurningOn)
                else -> error("Unexpected bluetooth state: $state")
            }
        }
        .onStart {
            val isEnabled = when (BluetoothAdapter.getDefaultAdapter()?.isEnabled) {
                true -> Available
                else -> Unavailable(reason = Off)
            }
            emit(isEnabled)
        }

internal actual val bluetoothAvailability: Flow<Bluetooth.Availability> =
    combine(
        locationEnabledFlow,
        bluetoothStateFlow,
    ) { locationEnabled, bluetoothState ->
        if (locationEnabled) bluetoothState else Unavailable(reason = LocationServicesDisabled)
    }
```

Reply:
Thank you for the review and especially for catching EXTRA_PROVIDER_ENABLE. I was too keen on code reduction. Also providing the flowof(true), I just did not find the appropriate implementation here, I tried a few thing and new I am still not there.

I think the name locationEnabledFlow is a bit missleading. I see the subject of this flow like -- 
 bluetoothAvailability does not need location to be enabled if sdk int is larger or equal  to android version R. 
I think naming is very important, but do not want to force something. Since I am not very good at naming.

Reply:
Maybe isLocationIndependent ?

Reply:
Let's leave the naming like you suggested it. 
Who is supposed to resolve the conversation. For me this one is resolved.

Reply:
> Who is supposed to resolve the conversation.

I'm actually not sure what standard convention is, but in this project I'm fine with either party resolving it. 🙂